include/rnp/rnp.h

```diff
@@ -171,10 +171,10 @@ typedef struct rnp_uid_handle_st *         rnp_uid_handle_t;
 typedef struct rnp_signature_handle_st *   rnp_signature_handle_t;
 
 /* Callbacks */
-typedef ssize_t rnp_input_reader_t(void *app_ctx, void *buf, size_t len);
-typedef void    rnp_input_closer_t(void *app_ctx);
-typedef bool    rnp_output_writer_t(void *app_ctx, const void *buf, size_t len);
-typedef void    rnp_output_closer_t(void *app_ctx, bool discard);
+typedef bool rnp_input_reader_t(void *app_ctx, void *buf, size_t len, size_t *read);
```

Comment:
Can we add some docs for this too? I think my understanding is that the reader should always return true unless an error occurs (so EOF would be indicated by returning true but setting *read=0). Let me know if I'm mistaken though.

Reply:
@dewyatt Agree, done. However now we seems to have problem with GTest...